pkg/util/openshift.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2021 Red Hat OpenShift Data Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func DetermineOpenShiftVersion (client client.Client) (string, error) {
+	// Determine ocp version
+	clusterVersionList := configv1.ClusterVersionList{}
+	if err := client.List(context.TODO(), &clusterVersionList); err != nil {
+		return "", err
+	}
+	clusterVersion := ""
+	for _, version := range clusterVersionList.Items {
+		clusterVersion = version.Status.Desired.Version
+	}
```

Comment:
What if this list is empty and this will send back "",nil which is not expected I guess. maybe one more check to check a empty value is required

Reply:
This list *shouldn't* ever be empty. If it is, then we have something dreadfully wrong. Maybe we should consider returning a more verbose message about this problem, but as it stands this is fine for now until if and when we ever run into this.